### DISCUSSION TITLE
Difference between Kong OSS and Kong Gateway Free

### DISCUSSION THREAD

**Discussion by monteiro-alan** [Kong/kong#9167] [Help] [answered]
Hello!

I'm using Kong OSS for a while, but since [Konga](https://github.com/pantsel/konga) is not receiving updates I'm a little afraid of continue using Kong OSS. I like the ideia of Kong Manager, but it is not clear for me what is the difference between Kong OSS and Kong Free, that comes with Kong Manager.

I appreciate any help in clarifying this doubt!

**Comment by chronolaw** [comment]
This link may give you some help. https://konghq.com/pricing.

In short Kong Free is Kong OSS + Kong Manager.

**Reply by MickBelker** [reply]
Does the OSS version have API gateway request limits?
If I look at the Kong Enterprise Free, they claim about "Up to 1M" requests per month.
1) Is there any hard limitation of API calls in OSS or Enterprise Free?
2) Is it possible / does it make sense to install Kong OSS, and then install Kong Manager standalone, configuring connection to Kong OSS?

I just need to evaluate whether I would run into any limits with the free version regarding management of the APIs / runtime of API calls. The API marketplace is not important for me, but management of the APIs from provider point of view.

**Reply by Yeganloo** [reply]
Anybody have any idea about these questions?
Is there any request/month restriction on self-managed OSS or Enterprise Free?
Kong documentation seems buggy. It claims that self-managed Enterprise Free dose not exists at all (but it dose!)
So i need some solid answer about this. i can not risk to find answer in action!

**Reply by sholaybhature** [reply]
https://docs.konghq.com/gateway/latest/#on-premises

**Reply by rhzs** [reply]
The doc is becoming unclear, I was coming from older OSS version of Kong 2.x. Looking at it now, it is becoming unclear, and more and more unfriendly for OSS.

**Reply by subnetmarco** [reply, marked as answer]
You can use Kong Gateway Free with no limitations, and it ships with Kong Manager. Effectively is a drop-in replacement for Kong Gateway OSS.

**Reply by tridoan** [reply]
Do you have a link to the git repo for this "Kong Gateway Free" version, that is not the Kong Gateway OSS version? Just so it is clear

**Reply by szmufo** [reply]
Kong Gateway Free where is the download link, Ubuntu Now i can only find the Enterprise version, thanks!

**Comment by khanakia** [comment]
Kong docs are so confusing there is no clear comparison between Open Source and Paid Version

**Reply by guanlan** [reply]
https://docs.konghq.com/gateway/latest/#features

**Comment by markwa2000** [comment]
Hello,

I have been referring many docs to understand the available deployment options in kubernetes using Helm charts. But these docs are so confusing with no proper explanation. In this https://docs.konghq.com/gateway/latest/#on-premises doc, It is mentioned that there are three ways to deploy self managed kong, 
1. kong OSS.
2. Kong Enterprise - 2.1. Free Enterprise & 2.2. Paid Enterprise.

In Github doc https://github.com/Kong/charts/blob/main/charts/kong/README.md#runtime-package , they mentioned that there are 3 packages to deploy in K8s
1. Kong Gateway (OpenSource)
2. Kong Enterprise K8S (No proper Explanation)
3. Kong Enterprise (No proper Explanation)

I tried to install the OpenSource Kong Gateway & I'm Getting this kind of DashBoard
<img width="960" alt="Screenshot 2024-02-13 124429" src="https://github.com/Kong/kong/assets/132652433/dced7cb3-09f6-4d0d-a59d-b22063e95842">

But when I check in the Kong Docs, the open source kong dashboard is like this 
![km_oss](https://github.com/Kong/kong/assets/132652433/7244762a-4ba3-40dc-8812-bc81be96d103)

How to download the open source kong manager using Helm Charts?

**Reply by Shamir-j** [reply]
Click on the default you should be able to see the same image

**Comment by ChrisRitchie** [comment]
I don't know what has happened to the docs, but most kong links above redirected to pages with similar (but not the same) content. Only two months ago I was following relatively clear examples on how to setup hybrid mode dataplane/control_plane. I have come back to it this week and lost hours of my time since all the links I had saved to docs are now redirected to really crappy docs, crappy with a crappy interface. The examples I used have now gone and documentation gone from OK-ish to total crap

**Reply by mheap** [reply]
Hey @ChrisRitchie! Do you have the URL that you were previously using? I'll figure out the best set of docs that show the same information and make sure the redirect is updated

**Reply by ChrisRitchie** [reply]
`https://docs.konghq.com/gateway/3.10.x/install/kubernetes/proxy/`
`https://docs.konghq.com/latest/configuration`

Above are a couple. I did find this below which has much clearer and easier to follow guidance:
`https://legacy-gateway--kongdocs.netlify.app/gateway-oss/2.5.x/hybrid-mode/`

The new docs seam to prefer hiding content behind panels which (for me at least) makes it a challenge to scan the page for content. Plus the list of links from the left hand side has bee removed and can now be found hidden behind the top menu drop downs. These links lead to a page of filtered How-To guides filtered by product rather than a concise guide EG:

`https://developer.konghq.com/how-to/?products=insomnia`

I think generally the change in format is a massive backward step. It kind of reminds me now of confluence, might look more pretty but in terms of ease of access and of information architecture, is all over the place and a nightmare to navigate.

**Reply by mheap** [reply]
https://developer.konghq.com/gateway/install/kubernetes/on-prem/ should work for you to install in hybrid mode on Kubernetes.

https://developer.konghq.com/gateway/configuration/ should work for https://docs.konghq.com/latest/configuration. I'll get this redirect fixed, thank you for bringing it to my attention.

Also, thank you for your feedback on the rework. We've done some user testing, but would love to do more. If you're willing to spend 45 minutes with the team we'd love to talk about how you use the docs and how we can improve them for your use case.

**Reply by youcandanch** [reply]
@mheap another one that seems to have fallen by the wayside is https://developer.konghq.com/gateway/#features, which was recommended in one of the discussions above about where to find the feature difference between what you can do in OSS vs. Enterprise. It _has_ been a little tricky to piece that together, especially as we're trying to figure out what's available in a _purely_ self-hosted OSS way; any recommendations on where that may live now?

**Reply by ChrisRitchie** [reply]
@mheap Thanks for your reply. The link you sent does appear to have what I was looking for. However it is no longer clear if OSS version exists any more. Do you need a license, what happens if you don't have a license? Any reference to the OSS version looks to have been removed.  As a sanity check, I did try to find that link you sent by navigating the menus but to no avail. 

There was version 3 documentation in the same format as this v2.5 documentation, but has been removed:
https://legacy-gateway--kongdocs.netlify.app/gateway-oss/2.5.x/hybrid-mode/

There was a page referencing best practices for an install on Kubernetes but I can no longer find this. 

For me a big issue is the removal a way to navigate logically grouped pages as was previously available via the side menu:
<img width="334" alt="Screenshot 2025-07-01 at 10 55 07" src="https://github.com/user-attachments/assets/7524e01e-fae7-45c3-98d3-b04b2b489e4d" />

The removal of a structured side menu, in favor of tiles and a global search on 'how to' docs does leave me feeling quite frustrated. I think a way is needed to regroup docs in some kind of logical flow again. And for a clear guide on using Kong without a licence (if this is even possible). I would imagine that it is the OSS users that are going to converted to Enterprise users as some point, if their application grows enough.
